src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts

```diff
@@ -232,10 +235,16 @@ export class MarkdownToolbarComponent extends AngularDisposable {
 			if (this.cellModel.currentMode !== CellEditModes.WYSIWYG) {
 				needsTransform = false;
 			} else {
+				let linkUrl = linkCalloutResult?.insertUnescapedLinkUrl;
+				const isFile = URI.parse(linkUrl).scheme === 'file';
+				if (isFile && !path.isAbsolute(linkUrl)) {
+					const notebookDirName = path.dirname(this.cellModel?.notebookModel?.notebookUri.fsPath);
+					const relativePath = (linkCalloutResult?.insertUnescapedLinkUrl).replace(/\\/g, path.posix.sep);
```

Comment:
linkCalloutResult is possibly undefined here yet you call replace without checking - which would throw. Is having an undefined result actually valid anyways?

Reply:
I think we check that it's not undefined before getting here

Reply:
https://github.com/microsoft/azuredatastudio/blob/main/src/sql/workbench/contrib/notebook/browser/cellViews/markdownToolbar.component.ts#L228

Reply:
If you're ever in this file again it would be nice to clean up that ? then since it's not doing anything

Reply:
you're right let me do that